projects/storefrontapp-e2e-cypress/cypress/integration/regression/pickup-in-store/pickup-checkout-journey.core-e2e-spec.ts

```diff
@@ -0,0 +1,170 @@
+          .should(
+            'be.equal',
+            interception.request.body.deliveryPointOfService.name
+          );
+      });
+
+      // Start of:- From the cart, the user can change the location they wish to pick up the product from.
+
+      cy.get(L.VIEW_CART).click();
+      cy.url().should('include', '/cart');
+      cy.get(L.PICKUP_STORE_LOCATION).should('be.visible');
+      cy.get(L.CHANGE_STORE_LINK).click();
+      cy.get(L.PICKUP_IN_STORE_MODAL).should('exist');
+
+      cy.intercept({
+        method: 'PATCH',
+        url: /users\/anonymous\/carts\/[0-9| a-z|-]*\/entries/,
+      }).as('changePickupInStore');
+
+      cy.get(L.ACTIVE_PICK_UP_IN_STORE_BUTTON).last().click();
+
+      cy.wait('@changePickupInStore').then((interception) => {
+        expect(interception.request.body).to.have.property(
+          'deliveryPointOfService'
+        );
+
+        // Start of - The user also add another item only for delivery.(Multiple items in cart)
+        cy.get(L.SAP_ICON_HOME_LINK).click();
+        cy.get(L.HOME_PAGE_SECOND_PRODUCT).click();
+        cy.get(L.ADD_TO_CART).click();
+        cy.get(L.VIEW_CART).click();
+        // End of - The user also add another item only for delivery.(Multiple items in cart)
+
+        //Start of:- The user decides to login so the order will show in the user's account.
+        cy.intercept({
+          method: 'POST',
+          url: '/authorizationserver/oauth/token',
+        }).as('registerUser');
+
+        register();
+
+        cy.wait('@registerUser').then((_interception) => {
+          login();
+        });
+        // End of:- The user decides to login so the order will show in the user's account.
+
+        // Start of:- The logged in user checks out.
+        cy.get(L.MINI_CART_BUTTON).click();
+        cy.get(L.PROCEED_TO_CHECKOUT_BUTTON).should('be.visible').click();
+
+        // Start of:- During checkout, the user fill address form.
+        fillAddressForm(defaultAddress);
+        cy.get(L.CHECKOUT_ADDRESS_FORM_SUBMIT_BUTTON).click();
+        // End of:- During checkout, the user fill address form.
+
+        // Start of:- During checkout, the user selects delivery mode.
+        cy.get(L.CHECKOUT_DELIVERY_MODE_CONTINUE_BUTTON)
+          .scrollIntoView()
+          .click({ force: true });
+        // End of:- During checkout, the user selects delivery mode.
+
+        // Start of:- During checkout, the user fill payment form.
+        fillPaymentForm(defaultPaymentDetails);
+        cy.get(L.CHECKOUT_PAYMENT_FORM_CONTINUE_BUTTON).click();
+        // End of:- During checkout, the user fill payment form.
+
+        //Start of:- order review
+        cy.get(L.CHANGE_STORE_LINK).first().click();
+        cy.get(L.PICKUP_IN_STORE_MODAL).should('exist');
+        cy.get(L.USE_MY_LOCATION).click();
+
+        const url = new RegExp(
+          `\/${EMAIL_ADDRESS.replace(
+            '@',
+            '%40'
+          )}\/carts\/[0-9| a-z''|-]*\/entries`,
```

Comment:
This regex seems to have some redundant terms in it. Let's have a call to go through it.

Reply:
Change this to `0-9a-zA-Z-`